assembly/src/parsers/u32_ops.rs

```diff
@@ -181,25 +181,82 @@ pub fn parse_u32split(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), A
 // ARITHMETIC OPERATIONS
 // ================================================================================================
 
-/// Translates u32add assembly instruction to VM operation U32ADD.
+/// Translates u32add assembly instruction to VM operations.
+/// Specifically handles these specific inputs per the spec.
+/// - Zero argument: Assert the top two elemenets are u32 and push the result after U32ADD to the stack
+/// - Single argument:
+///   - "unsafe" skips the assert check and direclty performs a U32ADD EQZ
+///   - "full" checks both numbers are u32 and perform the same operations as "unsafe"
+///   - Any number argument gets pushed to the stack, checked if both are u32 and performs a U32ADD.
```

Comment:
This seems out of date.

One other thing: could we add the count of number of VM cycles needed for each variant. Like `u32add.unsafe` requires just 1 VM cycles, `u32add.full` requires 9 VM cycles (if I did my count correctly) etc.